pkg/api/config.go

```diff
@@ -125,6 +129,40 @@ func validateBuildRootImageConfiguration(fieldRoot string, input *BuildRootImage
 	return validationErrors
 }
 
+func validateImages(fieldRoot string, input []ProjectDirectoryImageBuildStepConfiguration) []error {
+	var validationErrors []error
+	for num, image := range input {
+		fieldRootN := fmt.Sprintf("%s[%d]", fieldRoot, num)
+		if image.To == "" {
+			validationErrors = append(validationErrors, fmt.Errorf("%s: `to` must be set", fieldRootN))
+		}
+		if len(image.Substitute) > 0 {
+			if image.OperatorManifests == "" {
+				validationErrors = append(validationErrors, fmt.Errorf("%s.substitute: %s.operator_manifests must also be set", fieldRootN, fieldRootN))
+			}
+			for num, sub := range image.Substitute {
+				fieldRootN2 := fmt.Sprintf("%s.substitute[%d]", fieldRootN, num)
+				if sub.PullSpec == "" {
+					validationErrors = append(validationErrors, fmt.Errorf("%s.pullspec: must be set", fieldRootN2))
+				}
+				if sub.With == "" {
+					validationErrors = append(validationErrors, fmt.Errorf("%s.with: must be set", fieldRootN2))
+				}
+			}
+		}
+		if len(image.OperatorIndex) > 0 && image.OperatorManifests != "" {
```

Comment:
IDK this looks like more code and more work. Inferring is simple. Let's do it. I don't want to onboard teams to this.

Reply:
Yeah, I don't insist on allowing explicit. I never wanted people to actually use the explicit config.

Reply:
Added implicit generation of index image steps.